addons/website/controllers/main.py

```diff
@@ -333,6 +335,193 @@ def get_dynamic_snippet_templates(self, filter_name=False):
+        :type search_type: str
+        :param term: search term written by the user
+        :type term: str
+        :param order:
+        :type order: str
+        :param limit: number of results to consider, defaults to 5
+        :type limit: int
+        :param max_nb_chars: max number of characters for text fields
+        :type max_nb_chars: int
+        :param options: specific to search_type
+        :type options: dict containing
+            allowFuzzy: enables the fuzzy matching when truthy
+            fuzzy (boolean): True when called after finding a name through fuzzy matching
+
+        :returns: dict (or False if no result) containing
+            - 'results' (list): results (only their needed field values)
+                    note: the monetary fields will be strings properly formatted and
+                    already containing the currency
+            - 'results_count' (int): the number of results in the database
+                    that matched the search query
+            - 'parts' (dict): presence of fields across all results
+            - 'fuzzy_search': search term used instead of requested search
+        """
+        order = self._get_search_order(order)
+        options = options or {}
+        results_count, search_results, fuzzy_term = request.website._search_with_fuzzy(search_type, term, limit, order, options)
+        if results_count == 0:
+            return {
+                'results': [],
+                'results_count': 0,
+                'parts': {},
+            }
+        term = fuzzy_term or term
+        search_results = request.website._search_render_results(search_results, limit)
+
+        mappings = []
+        results_data = []
+        for search_result in search_results:
+            results_data += search_result['results_data']
+            mappings.append(search_result['mapping'])
+        if search_type == 'all':
+            # Only supported order for 'all' is on name
+            results_data.sort(key=lambda r: r.get('name', ''), reverse='name desc' in order)
+        results_data = results_data[:limit]
+        result = []
+        for record in results_data:
+            mapping = record['_mapping']
+            mapped = {
+                '_fa': record.get('_fa'),
+            }
+            for mapped_name, field_meta in mapping.items():
+                value = record.get(field_meta.get('name'))
+                if value:
+                    field_type = field_meta.get('type')
+                    if field_type == 'text':
+                        if value and len(value) > max_nb_chars:
+                            value = "%s..." % value[:(max_nb_chars - 3)]
+                        if field_meta.get('match') and value and term:
+                            pattern = '|'.join([re.escape(search_term) for search_term in term.split()])
+                            if pattern:
+                                parts = re.split('(%s)' % pattern, value, flags=re.I)
```

Comment:
```suggestion
                                parts = re.split(f'({pattern})', value, flags=re.IGNORECASE)
```
Good thinking on passing the flags by keyword, but you can also use the full flag name, it's a good habit.